src/types/user.ts

```diff
@@ -0,0 +1,77 @@
+import { Id } from "@/types/general";
+
+// This must always reflect the Rust struct on the backend entity::users::Model
+export interface User {
+  id: Id;
+  email: string;
+  password?: string;
+  first_name: string;
+  last_name: string;
+  display_name: string;
+}
+
+export interface NewUser {
+  email: string;
+  password?: string;
+  first_name: string;
+  last_name: string;
+  display_name: string;
+}
+
+export type UserCategory = "coach" | "coachee" | "generalUser";
```

Comment:
Nice, I like this.

Reply:
Is "generalUser" redundant perhaps since it's a User type anyway? Can we just use "general" instead?

Reply:
Good question. I actually removed this in 74a470fd25c121bdcaf12677819f4f2aa363f67b as it was a hold over from a previous implementation for this that I tried. 
I could see it coming back at some point though at which time I think you're right that the "User" part is redundant.